luthier/config/src/dimensions.lua

```diff
@@ -117,213 +122,226 @@ local FIELDSETS = {
 ]]
 -------------------------------------------------------------------------------
 
-local searchProviders = dimensionUtils.searchProviders
-local keyValueStores = dimensionUtils.keyValueStores
-
 return {
+    testDimension = {
+        longName = "a longName for testing",
+        description = "a description for testing",
+        fields = FIELDSETS.test,
+        category = "a category for testing",
+        type = "KeyValueStoreDimension",
+        isAggregatable = false,
+        defaultFields = {
```

Comment:
Keep in mind that the `defaultFields` is a subset of `fields` that are shown in
the response when the user doesn't say otherwise. Therefore, this should be a list of
strings that contains just the *names* of the fields to default to. So:
```lua
defaultFields = {"TEST_PK", "TEST_FIELD_1", "TEST_FIELD_2"
```
Note that the defaultFields are all already defined in `fields`.

Reply:
In that case, I will need to rewrite a parser in Java. Does it still make sense to take `defaultDimensionField` and build a LinkedHashSet of `DemensionField`? 
I think a list of Strings would be suffice.

Reply:
disregard this. Please use the comment two below to continue this conversation if need be.